pkg/adaptor/proxy/factory.go

```diff
@@ -0,0 +1,27 @@
+// Copyright Confidential Containers Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package proxy
+
+type Factory interface {
+	New(sandboxID string) AgentProxy
+}
+
+type factory struct {
+	PauseImage    string
+	CriSocketPath string
+}
+
+func NewFactory(pauseImage, criSocketPath string) Factory {
+	return &factory{
+		PauseImage:    pauseImage,
+		CriSocketPath: criSocketPath,
+	}
+}
+
+func (f *factory) New(socketPath string) AgentProxy {
```

Comment:
I think this should be named as `NewAgentProxy` because with plain `New` it signifies that we are creating new of the same object we are referring to. Which is kind of misleading.

And the actual `NewAgentProxy` can be made private because the individual cloud providers won't be calling this function any more. Infact the whole code can sit here.

Reply:
I agree.

NewAgentProxy is reference from cloud providers other than ibmcloud.   is reference from cloud providers other than ibmcloud. In this PR, I only refactored ibmcloud, and keep the other cloud providers unchanged.

I'll change the names of these functions, when we refactored all of the cloud providers.